src/customControls/ValueBasedCombo.java

```diff
@@ -0,0 +1,60 @@
+package customControls;
+
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+
+import org.eclipse.swt.widgets.Combo;
+
+public class ValueBasedCombo{
+	
+	private Combo comboBox;
+	private ArrayList<SimpleEntry<String, String>> keyValItemList;
+	
+	public ValueBasedCombo(Combo combo, ArrayList<SimpleEntry<String, String>> keyValItemList){
+		this.comboBox = combo;
+		this.keyValItemList = keyValListDeepCopy(keyValItemList);
```

Comment:
Nie ma potrzeby wykonywania głębokiej kopii listy par <k,v>; lista przekazana do konstruktora nawet po usunięciu nie spowoduje usunięcia obiektu listy bo pole this.keyValItemList jest referencją do tegoż obiektu.

Modyfikacja tej listy w klasie z której wywoływany jest konstruktor nie wchodzi w grę, gdyż obiekt (i podobiekty) jest/są wykożystywany/e tylko w operacjach odczytu.